Bottom: 27bfde2fb2ab4ee3033c6b465d55530a614e6bed
Top:    47c055283ab722851373ec3c7f8affac14e1f08b
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-25 15:04:52 -0700

rsocket: Set readfds event if rsocket has been disconnected

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/rsocket.c b/src/rsocket.c
index 394fed4..c833d46 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1631,7 +1631,7 @@ rs_poll_to_select(int nfds, struct pollfd *fds, fd_set *readfds,
 	int i, cnt = 0;
 
 	for (i = 0; i < nfds; i++) {
-		if (readfds && (fds[i].revents & POLLIN)) {
+		if (readfds && (fds[i].revents & (POLLIN | POLLHUP))) {
 			FD_SET(fds[i].fd, readfds);
 			cnt++;
 		}
